plc_ioc_status.py

```diff
@@ -0,0 +1,182 @@
+
+            widget = QtWidgets.QWidget()
+            widget_layout = QtWidgets.QHBoxLayout()
+
+            widget.setLayout(widget_layout)
+            widget.layout().addWidget(label_name)
+            widget.layout().addWidget(label_online)
+            widget.layout().addWidget(label_in_use)
+            widget.layout().addWidget(label_alarmed)
+            widget.layout().addWidget(label_heartbeat)
+            widget.layout().addWidget(plc_status_indicator)
+
+            self.plc_ioc_container.layout().addWidget(widget)
+        vertical_spacer = (
+            QtWidgets.QSpacerItem(20, 40,
+                                  QtWidgets.QSizePolicy.Preferred,
+                                  QtWidgets.QSizePolicy.MinimumExpanding))
+        self.plc_ioc_container.layout().addItem(vertical_spacer)
+
+    def heartbeat_severity_changed(self, key, alarm):
+        # 0 = NO_ALARM, 1 = MINOR, 2 = MAJOR, 3 = INVALID, # 4 = DISCONNECTED
+        plc = self.ffs_count_map.get(key)
+        if alarm == 3:
+            plc['plc_status'] = False
+        else:
+            plc['plc_status'] = True
+        self.update_status_labels(key)
+
+    def ffo_connection_callback(self, key, idx, conn):
+        # Update ffos count for connected In_Use PVs
+        plc = self.ffs_count_map.get(key)
+        plc['online'][idx] = conn
+        # Call routine to update proper label
+        self.update_plc_labels(key)
+
+    def ffo_value_changed(self, key, idx, value):
+        # Update ffos count for In_Use == True Pvs
+        plc = self.ffs_count_map.get(key)
+        plc['in_use'][idx] = value
+        self.update_plc_labels(key)
+
+    def ffo_severity_changed(self, key, idx, alarm):
+        # 0 = NO_ALARM, 1 = MINOR, 2 = MAJOR, 3 = INVALID, 4 = DISCONNECTED
+        plc = self.ffs_count_map.get(key)
+        # TODO: probably don't need severity at all, but if we do
+        if alarm != 0:
+            plc['alarmed'][idx] = True
+        else:
+            plc['alarmed'][idx] = False
+        self.update_plc_labels(key)
+
+    def update_plc_labels(self, key):
+        # Fetch value from count
+        # TODO maybe have some checks here....?
+        counts = self.ffs_count_map.get(key)
+        online_cnt = sum(counts['online'])
+        in_use_cnt = sum(counts['in_use'])
+        alarmed_cnt = sum(counts['alarmed'])
+        # Pick the label from the map
+        # Update label with new count
+        labels = self.ffs_label_map.get(key)
+        labels['online'].setText(str(online_cnt))
+        labels['in_use'].setText(str(in_use_cnt))
+        labels['alarmed'].setText(str(alarmed_cnt))
+
+    def update_status_labels(self, key):
+        status = self.ffs_count_map.get(key)
+        plc_status = status['plc_status']
+        labels = self.ffs_label_map.get(key)
+        if plc_status is True:
+            labels['plc_status'].setColor(self._on_color)
+        else:
+            labels['plc_status'].setColor(self._on_color)
```

Comment:
both branches of this if statement do the same thing

Reply:
oops thanks for pointing this out - will change this in a bit